### PR TITLE
Move Page functionality to modules

### PR DESCRIPTION
**Pull request by jillianrosile**

### CODE REVIEW

File: lib/gamera/modules/reloadable.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    # A method to wait for slow loading data on a page. Useful, for example,
+    # when waiting on a page that shows the count of records loaded via a slow
+    # web or import.
+    #
+    # @param retries [Integer] Number of times to reload the page before giving up
+    # @param allowed_errors [Array] Array of errors that trigger a refresh, e.g.,  if an ExpectationNotMetError was raised during an acceptance test
+    # @param block [Block] The block to execute after each refresh
+    def with_refreshes(retries, allowed_errors = [RSpec::Expectations::ExpectationNotMetError], &block)
+      retries_left ||= retries
+      visit
```
Comment (dobbs) [inline lib/gamera/modules/reloadable.rb]: `visit` is an implicit dependency on `Visitable`.  I'd like to make that explicit.

We could explicitly `include Gamera::Visitable`.  Or implement a `visit` method which raises an exception saying you need to implement a `visit` of your own or include ours.

We could also put `with_refreshes` in `Visitable`.  That's saying that we know visitable pages are reloadable.

I think I like the last option better, but I have to say I also like the separation of concerns expressed here in making it a separate module.

File: lib/gamera/modules/visitable.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    include Capybara::DSL
+
+    # Open the page url in the browser specified in your Capybara configuration
+    #
+    # @param fail_on_redirect [Boolean] Whether to fail if the site redirects to a page that does not match the url_matcher regex
+    # @raise [WrongPageVisited] if the site redirects to URL that doesn't match the url_matcher regex and fail_on_redirect is true
+    def visit(fail_on_redirect = true)
+      super(url)
+      if fail_on_redirect
+        fail WrongPageVisited, "Expected URL '#{url}', got '#{page.current_url}'" unless displayed?
```
Comment (dobbs) [inline lib/gamera/modules/visitable.rb]: Should this be `Gamera::WrongPageVisited` for consistency with the `fail` inside `displayed?` ?

File: lib/gamera/modules/visitable.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    # Methods to include on pages that are visitable, i.e. have a URL to visit and are then displayed.
+
+    include Capybara::DSL
+
+    # Open the page url in the browser specified in your Capybara configuration
+    #
+    # @param fail_on_redirect [Boolean] Whether to fail if the site redirects to a page that does not match the url_matcher regex
+    # @raise [WrongPageVisited] if the site redirects to URL that doesn't match the url_matcher regex and fail_on_redirect is true
+    def visit(fail_on_redirect = true)
+      super(url)
```
Comment (dobbs) [inline lib/gamera/modules/visitable.rb:12]: We should do something explicit with `url`.  Maybe `Visitable` wants an `attr_attribute :url`.  Or define a `url` method which raises a You-must-implement-this exception.

**Comment by jillianrosile** [comment]
@dobbs: I addressed your comments in my latest commit - let me know what you think.

**Comment by dobbs** [comment]
:+1:

**Comment by dobbs** [comment]
:+1:  (sorry for not getting back to you sooner :-)

**Pull request merged**